frontend/src/components/ConnectionStateIcon/ConnectionStateIcon.tsx

```diff
@@ -74,17 +84,54 @@ export function ConnectionStateIcon({
     element = (
       <span className={css.combo}>
         <Icon {...props} name="hdd" color="grayDarker" fixedWidth />
-        <sup>
-          <Icon name={icon} color={colorName} spin={state === 'connecting'} size="sm" type="regular" fixedWidth />
-        </sup>
+        {!(showQuality && device) && (
+          <sup>
+            <Icon name={icon} color={colorName} spin={state === 'connecting'} size="sm" type="regular" fixedWidth />
+          </sup>
+        )}
       </span>
     )
   else element = <Icon {...props} name={icon} color={colorName} spin={state === 'connecting'} fixedWidth />
 
+  if (showQuality && device) {
+    name = (
+      <>
+        <b className={css.capitalize}>{name}</b>
+        <br />
+        <Icon
+          name="circle"
+          color={device.quality === 'POOR' ? 'danger' : 'warning'}
+          size="bug"
+          type="solid"
+          inlineLeft
+        />
+        Connection quality {device.quality.toLowerCase()}
```

Comment:
if the `quality` is undefined, it is going to broke the component.

`device.quality.toLowerCase()`

Reply:
never mind 

you are checking 

`  let showQuality = device?.quality === 'POOR' || device?.quality === 'MODERATE'`

Reply:
could it be quality undefined?

Reply:
The typescript contract says it should never be undefined... so 🤷‍♂️ hopefully it speaks the truth!  🙂